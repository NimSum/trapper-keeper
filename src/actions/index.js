export const addNotes = notes => ({
  type: 'ADD_NOTES',
  notes
});

export const addNote = note => ({
  type: 'ADD_NOTE',
  note
})

export const updateNote = note => ({
  type: 'UPDATE_NOTE',
  note
})
