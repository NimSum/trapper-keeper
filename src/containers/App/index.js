import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addNotes } from '../../actions/';
import CardContainer from '../../components/CardContainer';
import { Route } from 'react-router-dom'
export class App extends Component {

  async componentDidMount() {
    const response = await fetch('http://localhost:3000/api/v1/notes');
    const notes = await response.json();
    this.props.setNotes(notes.notes)
  } 

  render() {
    return (
      <div>
        <h1>Trapper Keeper</h1>
        < CardContainer />
        <Route path='/new-note' component={Form} />
      </div>
    )
  }
}

const mapStateToProps = (state) => ({

})

const mapDispatchToProps = (dispatch) => ({
  setNotes: (notes) => dispatch(addNotes(notes))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
