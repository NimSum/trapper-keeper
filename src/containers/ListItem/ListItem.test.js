import React from 'react';
import ReactDOM from 'react-dom';
import  { ListItem }  from './index.js';
import { shallow } from 'enzyme';
import { mapDispatchToProps} from '../App'
import { addNotes } from '../../actions/index'

describe('ListItem', () => {
  let wrapper;
  let mockNotes;
  let mockItem = {
     id: 4,
     body: 'Hello', 
     completed: false
  }
  let mockEditing = false;
  let mockUpdateListItems = jest.fn()
  let mockDefaultState = {
    editable: false,
    body: ''
  } 

  beforeEach(()=> {
    wrapper = shallow(
    <ListItem
      item={ mockItem }
      editing={ mockEditing }
      updateListItems={ mockUpdateListItems }
    />, {disableLifecycleMethods: true})
  })

  it('should match the snapshot' , () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual(mockDefaultState)
  })

  it('should set the editable property to true', () => {
    wrapper.editItem = jest.fn()
    wrapper.instance().editItem()
    expect(wrapper.state()).toEqual({ body: "", editable: true })
  })

})


