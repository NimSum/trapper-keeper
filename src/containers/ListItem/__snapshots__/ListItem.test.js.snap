// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot 1`] = `
<li
  className="list-item"
  key="4"
>
  <div
    className="item-container"
  >
    <i
      className="far fa-square"
      onClick={[Function]}
    />
    <p
      className="list-item"
      onClick={false}
    >
      Hello
    </p>
    <i
      className="fas fa-times"
      onClick={[Function]}
    />
  </div>
</li>
`;
